diff -Naur linux-3.4.orig/drivers/misc/Kconfig linux-3.4.new/drivers/misc/Kconfig
--- linux-3.4.orig/drivers/misc/Kconfig	2012-04-22 00:47:52.000000000 +0300
+++ linux-3.4.new/drivers/misc/Kconfig	2012-04-26 12:49:49.550955923 +0300
@@ -498,6 +498,17 @@
 	  Maxim MAX8997 PMIC.
 	  The MAX8997 MUIC is a USB port accessory detector and switch.
 
+config MEMNOTIFY
+	tristate "Enables memory notification pseudo-device"
+	default m
+	help
+	  This option enables pseudo-device /dev/memnotify for tracking
+	  system memory utilization and updating state to subscribed clients
+	  when specified threshold reached.
+
+	  Say Y here if you want to support memory monitoring.
+	  If unsure, say N.
+
 source "drivers/misc/c2port/Kconfig"
 source "drivers/misc/eeprom/Kconfig"
 source "drivers/misc/cb710/Kconfig"
