diff --git a/libmemnotifyqt/debian/control b/libmemnotifyqt/debian/control
index 5a9ca36..ce086f9 100644
--- a/libmemnotifyqt/debian/control
+++ b/libmemnotifyqt/debian/control
@@ -2,34 +2,47 @@ Source: libmemnotifyqt
 Section: libs
 Priority: optional
 Maintainer: Leonid Moiseichuk <leonid.moiseichuk@nokia.com>
-Build-Depends: debhelper (>= 5), libqt4-dev (>= 4.5), libqt4-test (>= 4.5), doxygen, graphviz
-Standards-Version: 3.7.3
+Build-Depends: debhelper (>= 5), libqt4-dev (>= 4.5), libqt4-test (>= 4.5), doxygen, graphviz, cdbs, dejavu-fonts 
+Standards-Version: 3.8.0
 Description: Memory Notification library Qt-style.
+ The libmemnotifyqt implements a notification for memory usage events
+ according to levels specified in control file.
+
+Package: libmemnotifyqt0
+Section: libs
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}
+Description: Memory Notification library Qt-style.
+ The libmemnotifyqt implements a notification for memory usage events
+ according to levels specified in control file.
 
 Package: libmemnotifyqt-dev
 Section: libdevel
 Architecture: any
-Depends: libmemnotifyqt (= ${binary:Version}), libqt4-dev (>= 4.5)
+Depends: libmemnotifyqt0 (= ${binary:Version}), libqt4-dev (>= 4.5), ${misc:Depends}
 Recommends: libmemnotifyqt-doc
 Description: Memory Notification library Qt-style.
  Development package for Memory Notification.
 
-Package: libmemnotifyqt-dbg
+Package: libmemnotifyqt0-dbg
 Section: libdevel
 Priority: extra
 Architecture: any
-Depends: libmemnotifyqt (= ${binary:Version})
+Depends: libmemnotifyqt0 (= ${binary:Version}), ${misc:Depends}
 Description: Memory Notification library Qt-style. Debugging package.
+ Debug package for Memory Notification.
 
 Package: libmemnotifyqt-doc
 Section: doc
 Architecture: any
-Recommends: libmemnotifyqt, libmemnotifyqt-dev
+Recommends: libmemnotifyqt-dev (= ${binary:Version}), ${misc:Depends}
 Description: Memory Notification library Qt-style. Documentation for developers.
+ Documenation for Memory Notification.
 
 Package: libmemnotifyqt-test
 Section: libdevel
 Priority: extra
 Architecture: any
-Depends: libmemnotifyqt (= ${binary:Version}), ${shlibs:Depends}, ${misc:Depends}
+Depends: libmemnotifyqt0 (= ${binary:Version}), ${shlibs:Depends}, ${misc:Depends}
 Description: Memory Notification library Qt-style. Testing files.
+ Tests for Memory Notification.
diff --git a/libmemnotifyqt/debian/libmemnotifyqt-dev.dirs b/libmemnotifyqt/debian/libmemnotifyqt-dev.dirs
index b843f92..47f8ca3 100644
--- a/libmemnotifyqt/debian/libmemnotifyqt-dev.dirs
+++ b/libmemnotifyqt/debian/libmemnotifyqt-dev.dirs
@@ -1,2 +1,3 @@
 usr/lib
 usr/include/memnotify
+usr/lib/pkgconfig
diff --git a/libmemnotifyqt/debian/libmemnotifyqt-dev.install b/libmemnotifyqt/debian/libmemnotifyqt-dev.install
index d23d299..78f4f14 100644
--- a/libmemnotifyqt/debian/libmemnotifyqt-dev.install
+++ b/libmemnotifyqt/debian/libmemnotifyqt-dev.install
@@ -1,2 +1,4 @@
 usr/include/memnotify/*
 usr/lib/libmemnotifyqt.so
+usr/lib/pkgconfig/libmemnotifyqt.pc
+
diff --git a/libmemnotifyqt/debian/libmemnotifyqt-doc.dirs b/libmemnotifyqt/debian/libmemnotifyqt-doc.dirs
index 3593c5d..246dfd9 100644
--- a/libmemnotifyqt/debian/libmemnotifyqt-doc.dirs
+++ b/libmemnotifyqt/debian/libmemnotifyqt-doc.dirs
@@ -1 +1 @@
-usr/share/doc/libmemnotifyqt
+usr/share/doc/libmemnotifyqt-doc
diff --git a/libmemnotifyqt/debian/libmemnotifyqt-doc.install b/libmemnotifyqt/debian/libmemnotifyqt-doc.install
index 2348812..246dfd9 100644
--- a/libmemnotifyqt/debian/libmemnotifyqt-doc.install
+++ b/libmemnotifyqt/debian/libmemnotifyqt-doc.install
@@ -1 +1 @@
-usr/doc/* usr/share/doc/libmemnotifyqt
+usr/share/doc/libmemnotifyqt-doc
diff --git a/libmemnotifyqt/debian/libmemnotifyqt-test.dirs b/libmemnotifyqt/debian/libmemnotifyqt-test.dirs
index 203bad3..18684a5 100644
--- a/libmemnotifyqt/debian/libmemnotifyqt-test.dirs
+++ b/libmemnotifyqt/debian/libmemnotifyqt-test.dirs
@@ -1,2 +1 @@
 usr/lib/libmemnotifyqt/tests
-usr/share/libmemnotifyqt/testdata
diff --git a/libmemnotifyqt/debian/libmemnotifyqt-test.install b/libmemnotifyqt/debian/libmemnotifyqt-test.install
index 44e67f9..18684a5 100644
--- a/libmemnotifyqt/debian/libmemnotifyqt-test.install
+++ b/libmemnotifyqt/debian/libmemnotifyqt-test.install
@@ -1,3 +1 @@
-usr/tests/tst_* usr/lib/libmemnotifyqt/tests
-usr/tests/run_libmemnotifyqt.sh usr/lib/libmemnotifyqt/tests
-usr/tests/testdata/* /usr/share/libmemnotifyqt
+usr/lib/libmemnotifyqt/tests
diff --git a/libmemnotifyqt/debian/libmemnotifyqt0.dirs b/libmemnotifyqt/debian/libmemnotifyqt0.dirs
new file mode 100644
index 0000000..6845771
--- /dev/null
+++ b/libmemnotifyqt/debian/libmemnotifyqt0.dirs
@@ -0,0 +1 @@
+usr/lib
diff --git a/libmemnotifyqt/debian/libmemnotifyqt0.install b/libmemnotifyqt/debian/libmemnotifyqt0.install
new file mode 100644
index 0000000..6337ab0
--- /dev/null
+++ b/libmemnotifyqt/debian/libmemnotifyqt0.install
@@ -0,0 +1 @@
+usr/lib/libmemnotifyqt.so.*
diff --git a/libmemnotifyqt/debian/rules b/libmemnotifyqt/debian/rules
index c2d6072..a5d2377 100755
--- a/libmemnotifyqt/debian/rules
+++ b/libmemnotifyqt/debian/rules
@@ -1,33 +1,11 @@
 #!/usr/bin/make -f
 
-# Uncomment this to turn on verbose mode.
 export DH_VERBOSE=1
 
-# shared library versions, option 1
-#version=2.0.5
-#major=2
-# option 2, assuming the library is created as src/.libs/libfoo.so.2.0.5 or so
-#version=`ls src/.libs/lib*.so.* | \
-# awk '{if (match($$0,/[0-9]+\.[0-9]+\.[0-9]+$$/)) print substr($$0,RSTART)}'`
-#major=`ls src/.libs/lib*.so.* | \
-# awk '{if (match($$0,/\.so\.[0-9]+$$/)) print substr($$0,RSTART+4)}'`
-
 include /usr/share/cdbs/1/rules/debhelper.mk
-include /usr/share/cdbs/1/class/makefile.mk
-# include /usr/share/cdbs/1/rules/patchsys-quilt.mk
-include /usr/share/cdbs/1/rules/utils.mk
+include /usr/share/cdbs/1/class/qmake.mk
 
-DEB_MAKE_INVOKE := $(MAKE) QMAKE_CFLAGS_DEBUG="-g1 -O0"
-DEB_MAKE_INSTALL_TARGET := INSTALL_ROOT=$(DEB_DESTDIR) install
 DEB_DH_INSTALL_SOURCEDIR := debian/tmp
 
-common-configure-arch:: configure-stamp generate_docs
-configure-stamp:
-	qmake QMF_INSTALL_ROOT=/usr CONFIG+="$(EXTRA_QMAKE_CONFIG_PARAMS)"  QMAKE_STRIP=/bin/true QMAKE_CXXFLAGS="-g1"
-	touch $@
-
-generate_docs:
-	cd doc && doxygen
-	mkdir -p $(DEB_DH_INSTALL_SOURCEDIR)/usr/doc && cp -R doc/html $(DEB_DH_INSTALL_SOURCEDIR)/usr/doc
-
-
+clean::
+	rmdir doc/html src/{release,debug} ||:
diff --git a/libmemnotifyqt/doc/Doxyfile b/libmemnotifyqt/doc/Doxyfile
index 667d996..557b4bb 100644
--- a/libmemnotifyqt/doc/Doxyfile
+++ b/libmemnotifyqt/doc/Doxyfile
@@ -85,9 +85,9 @@ WARN_LOGFILE           =
 #---------------------------------------------------------------------------
 # configuration options related to the input files
 #---------------------------------------------------------------------------
-INPUT                  = ../src/plugins
+INPUT                  = ../src
 INPUT_ENCODING         = UTF-8
-FILE_PATTERNS          = *.cpp *.h
+FILE_PATTERNS          = *.cpp *.hpp
 RECURSIVE              = YES
 EXCLUDE                = 
 EXCLUDE_SYMLINKS       = NO
diff --git a/libmemnotifyqt/doc/doc.pro b/libmemnotifyqt/doc/doc.pro
index 9acfb8f..18b2e57 100644
--- a/libmemnotifyqt/doc/doc.pro
+++ b/libmemnotifyqt/doc/doc.pro
@@ -1,13 +1,12 @@
 dox.target = doc
-contains(CONFIG, nodoc) {
-dox.commands = mkdir -p $$PWD/html && touch $$PWD/html/not_docs_here_yet
-}
-else {
-dox.commands = doxygen $$PWD/Doxyfile
-}
+dox.commands = doxygen
+dox.files = html/* 
+dox.path  = /usr/share/doc/libmemnotifyqt-doc/html
 dox.depends =
-dox.files = html
-dox.path  = $$ASD_INSTALL_ROOT/share/doc/libmemnotifyqt
-QMAKE_EXTRA_TARGETS += dox
 
-INSTALLS += dox
+texts.files = design.txt environment.txt
+texts.path = /usr/share/doc/libmemnotifyqt-doc
+texts.depends = 
+
+QMAKE_EXTRA_TARGETS += dox
+INSTALLS += dox texts
diff --git a/libmemnotifyqt/libmemnotifyqt.pro b/libmemnotifyqt/libmemnotifyqt.pro
index f04ee78..cb2183e 100644
--- a/libmemnotifyqt/libmemnotifyqt.pro
+++ b/libmemnotifyqt/libmemnotifyqt.pro
@@ -3,12 +3,13 @@ TEMPLATE = subdirs
 SUBDIRS = \
           doc \
           src \
-#          test
+          test
 
-CONFIG += ordered
-
-# Make it so projects can find our specific features
-system(echo QMAKEFEATURES+=$$PWD/features > $$OUT_PWD/.qmake.cache)
+CONFIG += release
 
 QMAKE_CLEAN += configure-stamp
 
+pc.files = libmemnotifyqt.pc
+pc.path = /usr/lib/pkgconfig
+
+INSTALLS += pc
diff --git a/libmemnotifyqt/src/src.pro b/libmemnotifyqt/src/src.pro
index b656daa..dab0b5d 100644
--- a/libmemnotifyqt/src/src.pro
+++ b/libmemnotifyqt/src/src.pro
@@ -18,6 +18,12 @@ SOURCES += debug_watcher.cpp
 SOURCES += poller.cpp 
 SOURCES += memory_notification.cpp 
 DEFINES += QT_DLL MEMNOTIFY_SHARED MEMNOTIFY_VERSION=\\\"$${VERSION}\\\"
-CONFIG  += qt warn_on debug_and_release build_all
+CONFIG  += qt warn_on release build_all
 QT -= gui
 
+target.path = /usr/lib
+
+includes.path = /usr/include
+includes.files = memnotify
+
+INSTALLS += includes target
diff --git a/libmemnotifyqt/test/cached_file/cached_file.pro b/libmemnotifyqt/test/cached_file/cached_file.pro
index 7b3e865..a0210b6 100644
--- a/libmemnotifyqt/test/cached_file/cached_file.pro
+++ b/libmemnotifyqt/test/cached_file/cached_file.pro
@@ -8,3 +8,7 @@ SOURCES += test-cached_file.cpp
 DEFINES += MEMNOTIFY_VERSION=\\\"$${VERSION}\\\"
 CONFIG  += qt warn_on debug build_all
 QT -= gui
+
+target.path = /usr/lib/libmemnotifyqt/tests
+
+INSTALLS += target
diff --git a/libmemnotifyqt/test/memory_notification/memory_notification.pro b/libmemnotifyqt/test/memory_notification/memory_notification.pro
index fc6c3b8..2cdf884 100644
--- a/libmemnotifyqt/test/memory_notification/memory_notification.pro
+++ b/libmemnotifyqt/test/memory_notification/memory_notification.pro
@@ -21,3 +21,10 @@ SOURCES += test-memory_notification.cpp
 DEFINES += MEMNOTIFY_VERSION=\\\"$${VERSION}\\\"
 CONFIG  += qt warn_on debug build_all
 QT -= gui
+
+target.path = /usr/lib/libmemnotifyqt/tests
+
+extra.files = debug.mn cgroup.nm sensor.txt
+extra.path = /usr/lib/libmemnotifyqt/tests
+
+INSTALLS += target extra
diff --git a/libmemnotifyqt/test/platform/platform.pro b/libmemnotifyqt/test/platform/platform.pro
index c6f2df3..bf3806d 100644
--- a/libmemnotifyqt/test/platform/platform.pro
+++ b/libmemnotifyqt/test/platform/platform.pro
@@ -8,3 +8,7 @@ SOURCES += test-platform.cpp
 DEFINES += MEMNOTIFY_VERSION=\\\"$${VERSION}\\\"
 CONFIG  += qt warn_on debug build_all
 QT -= gui
+
+target.path = /usr/lib/libmemnotifyqt/tests
+
+INSTALLS += target
diff --git a/libmemnotifyqt/test/test.pro b/libmemnotifyqt/test/test.pro
index 44fc9d6..a264ad0 100644
--- a/libmemnotifyqt/test/test.pro
+++ b/libmemnotifyqt/test/test.pro
@@ -4,7 +4,3 @@ SUBDIRS       = \
 	platform \
 	watcher_builder \
 	memory_notification \
-
-# install
-#INSTALLS += target 
-
diff --git a/libmemnotifyqt/test/watcher_builder/watcher_builder.pro b/libmemnotifyqt/test/watcher_builder/watcher_builder.pro
index 1e2d971..cb90199 100644
--- a/libmemnotifyqt/test/watcher_builder/watcher_builder.pro
+++ b/libmemnotifyqt/test/watcher_builder/watcher_builder.pro
@@ -12,3 +12,7 @@ SOURCES += test-watcher_builder.cpp
 DEFINES += MEMNOTIFY_VERSION=\\\"$${VERSION}\\\"
 CONFIG  += qt warn_on debug build_all
 QT -= gui
+
+target.path = /usr/lib/libmemnotifyqt/tests
+
+INSTALLS += target
